Unable to build latest with Mono for X11

### Godot version

efc87481e45c66b3a18dfe1db82c3289fe2a370d

### System information

Ubuntu 20.04

### Issue description

I'm attempting to compile the latest version following the [documentation here](https://docs.godotengine.org/en/stable/development/compiling/compiling_with_mono.html) and getting stuck with the compilation error below.

I can compile successfully by checking out 3.3.3 stable plus the extra commit I'm interested in with `git cherry-pick`: https://github.com/godotengine/godot/commit/6db17a523e3747c489747d52de62790d9148d25d

Apologies in advance if there's an obvious reason for this, as I'm not familiar with C++ or its build tools. I believe I've satisfied all the dependencies listed [here](https://docs.godotengine.org/en/stable/development/compiling/compiling_for_x11.html) and am at a dead end with this. My compiler is GCC 9.3.0.

```
scons: Reading SConscript files ...
Platform "x11" has been renamed to "linuxbsd" in Godot 4.0. Building for platform "linuxbsd".
Enabling ALSA
Enabling PulseAudio
D-Bus development libraries not found, disabling dependent features
Enabling udev support
Note: Building a debug binary (which will run slowly). Use `target=release_debug` to build an optimized release binary.

scons: warning: Calling missing SConscript without error is deprecated.
Transition by adding must_exist=0 to SConscript calls.
Missing SConscript 'modules/gdnative/xr/SCsub'
File "/home/censored/Documents/projects/godot/modules/gdnative/SCsub", line 20, in <module>
Mono root directory not found. Using pkg-config instead
Checking for C header file mntent.h... (cached) yes
scons: done reading SConscript files.
scons: Building targets ...
[ 40%] Compiling ==> modules/mono/editor/code_completion.cpp
modules/mono/editor/code_completion.cpp: In function 'PackedStringArray gdmono::get_code_completion(gdmono::CompletionKind, const String&)':
modules/mono/editor/code_completion.cpp:126:73: error: 'begin' was not declared in this scope; did you mean 'std::begin'?
  126 |     for (const KeyValue<StringName, ProjectSettings::AutoloadInfo> &E : autoloads) {
      |                                                                         ^~~~~~~~~
      |                                                                         std::begin
In file included from /usr/include/c++/9/string:54,
                 from /usr/include/c++/9/stdexcept:39,
                 from /usr/include/c++/9/array:39,
                 from /usr/include/c++/9/tuple:39,
                 from /usr/include/c++/9/mutex:38,
                 from ./core/os/mutex.h:39,
                 from ./core/string/string_name.h:34,
                 from ./core/string/node_path.h:34,
                 from ./core/variant/variant.h:46,
                 from modules/mono/editor/code_completion.h:35,
                 from modules/mono/editor/code_completion.cpp:31:
/usr/include/c++/9/bits/range_access.h:105:37: note: 'std::begin' declared here
  105 |   template<typename _Tp> const _Tp* begin(const valarray<_Tp>&);
      |                                     ^~~~~
modules/mono/editor/code_completion.cpp:126:73: error: 'end' was not declared in this scope; did you mean 'std::end'?
  126 |     for (const KeyValue<StringName, ProjectSettings::AutoloadInfo> &E : autoloads) {
      |                                                                         ^~~~~~~~~
      |                                                                         std::end
In file included from /usr/include/c++/9/string:54,
                 from /usr/include/c++/9/stdexcept:39,
                 from /usr/include/c++/9/array:39,
                 from /usr/include/c++/9/tuple:39,
                 from /usr/include/c++/9/mutex:38,
                 from ./core/os/mutex.h:39,
                 from ./core/string/string_name.h:34,
                 from ./core/string/node_path.h:34,
                 from ./core/variant/variant.h:46,
                 from modules/mono/editor/code_completion.h:35,
                 from modules/mono/editor/code_completion.cpp:31:
/usr/include/c++/9/bits/range_access.h:107:37: note: 'std::end' declared here
  107 |   template<typename _Tp> const _Tp* end(const valarray<_Tp>&);
      |                                     ^~~
scons: *** [modules/mono/editor/code_completion.linuxbsd.tools.64.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:03.641]
```

Possibly something about https://github.com/godotengine/godot/pull/52122?

### Steps to reproduce

Attempt to compile latest on Ubuntu 20.04.

### Minimal reproduction project

_No response_

--- a/modules/mono/editor/code_completion.cpp
+++ b/modules/mono/editor/code_completion.cpp
@@ -123,8 +123,8 @@ PackedStringArray get_code_completion(CompletionKind p_kind, const String &p_scr
 				// AutoLoads
 				OrderedHashMap<StringName, ProjectSettings::AutoloadInfo> autoloads = ProjectSettings::get_singleton()->get_autoload_list();
 
-				for (const KeyValue<StringName, ProjectSettings::AutoloadInfo> &E : autoloads) {
-					const ProjectSettings::AutoloadInfo &info = E.value;
+				for (OrderedHashMap<StringName, ProjectSettings::AutoloadInfo>::Element E = autoloads.front(); E; E = E.next()) {
+					const ProjectSettings::AutoloadInfo &info = E.value();
 					suggestions.push_back(quoted("/root/" + String(info.name)));
 				}
 			}

